devices/blepsynth/laddervcf.hh

```diff
@@ -171,10 +178,18 @@ private:
       reso += drive * sqrt (reso) * reso * 0.03f;
 
     float vol = exp2f ((drive + -12 * sqrt (reso)) * db_x2_factor);
-    fparams.pre_scale = negative_drive_vol * vol;
-    fparams.post_scale = std::max (1 / vol, 1.0f);
+    fparams.pre_scale = negative_drive_vol * vol * global_volume_;
+    fparams.post_scale = std::max (1 / vol, 1.0f) / global_volume_;
     fparams.reso = sqrt (reso) * 4;
   }
+  static float
+  tanh_approx (float x)
+  {
+    // https://www.musicdsp.org/en/latest/Other/238-rational-tanh-approximation.html
+    x = std::clamp (x, -3.0f, 3.0f);
+
+    return x * (27.0f + x * x) / (27.0f + 9.0f * x * x);
+  }
```

Comment:
Wow, I didn't really know what kind of distortion the old saturation code causes. I have plotted the curves here:

![newsat](https://github.com/tim-janik/anklang/assets/281887/f7481176-035b-490c-8eb2-98140785b51f)


Would just be interested to know how badly the DIV affects performance...

Reply:
In dsp-research you can measure that with testfilter perf lp2 (or similar).

Sallen-Key Filter 2nd order with tanh approx with division, stereo., no freq/reso modulation in block:

with mono=0 freq=0 reso=0: ns/sample 81.855245
                              bogopolyphony = 254.514336

with polynomial saturation

with mono=0 freq=0 reso=0: ns/sample 70.393551
                              bogopolyphony = 295.955141

So yes, it costs us some ns/sample (some bogovoices), but its not catastrophically expensive. Of course this depends on filter type (Sallen-Key vs. Ladder) and filter mode.